compiler/rustc_mir/src/interpret/step.rs

```diff
@@ -113,6 +113,37 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 M::retag(self, *kind, dest)?;
             }
 
+            // Call CopyNonOverlapping
+            CopyNonOverlapping(box rustc_middle::mir::CopyNonOverlapping { dst, src, count }) => {
+                let count = self.eval_operand(count, None)?;
+                let count = self.read_scalar(count)?.to_machine_usize(self)?;
+
+                let src = self.eval_operand(src, None)?;
+                let dst = self.eval_operand(dst, None)?;
+                let layout =
+                    src.layout.pointee_info_at(self, rustc_target::abi::Size::ZERO).unwrap();
```

Comment:
@oli-obk any idea if the `unwrap` here could become a problem? Looks like `pointee_info_at` "masks" layout computation errors and turns them into `Option` so it's tricky to re-raise the original problem here.

Reply:
I don't think we can hit problems right now, and I'm not sure if it's possible to hit any problems ever without bailing out earlier anyway. But I would prefer this code to unconditionally use `src.layout.ty.builtin_deref(true).unwrap()` and compute the layout of that. We know that the argument must be a pointer, so we can safely unwrap the `builtin_deref`.